### PR TITLE
Add http head requests

### PR DESCRIPTION
**Pull request by briandowns**
### What's Changed:

Adds the ability to send and process HTTP HEAD requests.

### Type of Change:

- [ ] Bug fix
- [ X ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping:

- [ X ] Tests have been updated to reflect the changes done within this PR (if applicable).
- [ X ] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

File: docs/docs/standard-lib/http.md [hunk 1]
```diff
@@ -112,9 +125,20 @@ httpClient.put("https://httpbin.org/put");
...
+```cs
+httpClient.head("https://httpbin.org/get");
+
+{"content": "", "headers": ["...", "..."], "statusCode": 200}
+```
+
 ### Response
 
-Both HTTP.get(), HTTP.post(), httpClient.get(), and httpClient.post() return a Result that unwraps a Response object on success, or nil on error.
+Both HTTP.get(), HTTP.post(), httpClient.put(), and httpClient.head() return a Result that unwraps a Response object on success, or nil on error.
```
Comment (Jason2605) [inline docs/docs/standard-lib/http.md:141]: Probably worth just stating all of them return the response

```suggestion
All HTTP requests return a Result that unwraps a Response object on success, or nil on error.
```

**Review by Jason2605 — COMMENTED** [review]
Small comment

**Comment by briandowns** [comment]
All set. Thanks!

**Comment by Jason2605** [comment]
Thanks for this!

**Pull request merged**